src/Microsoft.Azure.WebJobs.Host/Scale/TriggerMetadata.cs

```diff
@@ -0,0 +1,41 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Azure.WebJobs.Host.Scale
+{
+    /// <summary>
+    /// Represents trigger metadata.
+    /// </summary>
+    public class TriggerMetadata
+    {
+        public TriggerMetadata(string functionName, JObject metadata)
```

Comment:
[Mathew]: I revamped the trigger metadata / provider APIs to make them cleaner. We may need to iterate more.

Reply:
I added the logic to get function name and type directly from metadata